### ISSUE TITLE
Duplication

### ISSUE THREAD

**Issue by neginsadeghi** [scmanjarrez/CVEScannerV2#6]
Hello there, I don't know why but I get the cvescannerv2 output with duplication, why it mentions the CVEs three times ? 

root@ubuntu20:/usr/share/nmap/scripts# nmap  -sV --script default,banner-plus.nse,ssl-cert,ssl-date,ssl-enum-ciphers,cvescannerv2 xxx -p 443,80,444 -Pn
Starting Nmap 7.80 ( https://nmap.org ) at 2023-09-05 13:50 +0330
NSE: [cvescannerv2] Timestamp: 2023-09-05T10:20:45+00:00
NSE: [cvescannerv2] CVE data source: nvd.nist.gov
NSE: [cvescannerv2] Script version: 3.0
Nmap scan report for xxxx
Host is up (0.085s latency).

PORT    STATE SERVICE  VERSION
80/tcp  open  http     nginx 1.18.0 (Ubuntu)

| cvescannerv2:
|   product: nginx
|   version: 1.18.0
|   vupdate: *
|   cves: 4
|       CVE ID                  CVSSv2  CVSSv3  ExploitDB       Metasploit
|       CVE-2022-41741          nil     7.8     No              No
|       CVE-2021-23017          6.8     7.7     No              No
|       CVE-2021-3618           5.8     7.4     No              No
|       CVE-2022-41742          nil     7.1     No              No
|
|   product: nginx
|   version: 1.18.0
|   vupdate: *
|   cves: 4
|       CVE ID                  CVSSv2  CVSSv3  ExploitDB       Metasploit
|       CVE-2022-41741          nil     7.8     No              No
|       CVE-2021-23017          6.8     7.7     No              No
|       CVE-2021-3618           5.8     7.4     No              No
|       CVE-2022-41742          nil     7.1     No              No
|
|   product: nginx
|   version: 1.18.0
|   vupdate: *
|   cves: 4
|       CVE ID                  CVSSv2  CVSSv3  ExploitDB       Metasploit
|       CVE-2022-41741          nil     7.8     No              No
|       CVE-2021-23017          6.8     7.7     No              No
|       CVE-2021-3618           5.8     7.4     No              No
|       CVE-2022-41742          nil     7.1     No              No

443/tcp open  ssl/http nginx 1.18.0 (Ubuntu)

| cvescannerv2:
|   product: nginx
|   version: 1.18.0
|   vupdate: *
|   cves: 4
|       CVE ID                  CVSSv2  CVSSv3  ExploitDB       Metasploit
|       CVE-2022-41741          nil     7.8     No              No
|       CVE-2021-23017          6.8     7.7     No              No
|       CVE-2021-3618           5.8     7.4     No              No
|       CVE-2022-41742          nil     7.1     No              No
|
|   product: nginx
|   version: 1.18.0
|   vupdate: *
|   cves: 4
|       CVE ID                  CVSSv2  CVSSv3  ExploitDB       Metasploit
|       CVE-2022-41741          nil     7.8     No              No
|       CVE-2021-23017          6.8     7.7     No              No
|       CVE-2021-3618           5.8     7.4     No              No
|       CVE-2022-41742          nil     7.1     No              No
|
|   product: nginx
|   version: 1.18.0
|   vupdate: *
|   cves: 4
|       CVE ID                  CVSSv2  CVSSv3  ExploitDB       Metasploit
|       CVE-2022-41741          nil     7.8     No              No
|       CVE-2021-23017          6.8     7.7     No              No
|       CVE-2021-3618           5.8     7.4     No              No
|       CVE-2022-41742          nil     7.1     No              No
|_

444/tcp open  ssl/http nginx 1.18.0 (Ubuntu)

| cvescannerv2:
|   product: nginx
|   version: 1.18.0
|   vupdate: *
|   cves: 4
|       CVE ID                  CVSSv2  CVSSv3  ExploitDB       Metasploit
|       CVE-2022-41741          nil     7.8     No              No
|       CVE-2021-23017          6.8     7.7     No              No
|       CVE-2021-3618           5.8     7.4     No              No
|       CVE-2022-41742          nil     7.1     No              No
|
|   product: nginx
|   version: 1.18.0
|   vupdate: *
|   cves: 4
|       CVE ID                  CVSSv2  CVSSv3  ExploitDB       Metasploit
|       CVE-2022-41741          nil     7.8     No              No
|       CVE-2021-23017          6.8     7.7     No              No
|       CVE-2021-3618           5.8     7.4     No              No
|       CVE-2022-41742          nil     7.1     No              No
|
|   product: nginx
|   version: 1.18.0
|   vupdate: *
|   cves: 4
|       CVE ID                  CVSSv2  CVSSv3  ExploitDB       Metasploit
|       CVE-2022-41741          nil     7.8     No              No
|       CVE-2021-23017          6.8     7.7     No              No
|       CVE-2021-3618           5.8     7.4     No              No
|       CVE-2022-41742          nil     7.1     No              No
|_
|

Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
Nmap done: 1 IP address (1 host up) scanned in 68.58 seconds

![image](https://github.com/scmanjarrez/CVEScannerV2/assets/85784596/cff85b7e-e275-4447-91c7-69b5e58db47b)

**Comment by scmanjarrez** [comment]
Can you upload the files generated by this command?

```
nmap -sV -n --script cvescannerv2 --script-args log=test.log,json=test.json -oN test -d -vv your_ip > test.txt 2>&1 
```

**Comment by neginsadeghi** [comment]
Yeah, here you go

[tests.zip](https://github.com/scmanjarrez/CVEScannerV2/files/12523436/tests.zip)

**Comment by scmanjarrez** [comment]
I see, the problem here is that I'm "adding" the data to be output in nmap on the fly (because each port is treated individually, but the list is the same", so at the end of the scan, it prints the final result (merged list). I'll have to rethink how to implement this trying to avoid duplicates.

**Comment by scmanjarrez** [comment]
Fixed in 689d6fd8dc2cc96e5203baa29442004cfb35755f, can you check that it indeed works? (I've tested it myself and seems to work)
